python/cuml/cuml/svm/svc.pyx

```diff
@@ -42,6 +44,54 @@ from libcpp cimport nullptr
 from cuml.svm.svm_base import SVMBase
 from cuml.internals.import_utils import has_sklearn
 from cuml.internals.array_sparse import SparseCumlArray
+from cuml.internals.api_decorators import device_interop_preparation, enable_device_interop
+
+from sklearn.svm import SVC as skSVC
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.multiclass import OneVsRestClassifier, OneVsOneClassifier
+from cuml.internals.mem_type import MemoryType
+from cuml.internals.available_devices import is_cuda_available
+
+
+class cpuModelSVC(skSVC):
```

Comment:
Why go with this mixture of inheritance (from `sklearn.svm.SVC`) and composition (constructing and storing an estimator in an instance attribute)?

How about something like the following sketch:
```python
# Not sure if this is the correct base class to inherit from
class CPUModel(BaseEstimator):
  def fit(self, X, y):
    params = self.get_params()
    if self.probability:
      self._estimator = sklearn.svm.SVC(**params)
    elif ...:
      # needs more `if` statements to handle the ovr and ovo cases, etc
      self._estimator = CalibratedClassifierCV(OneVsRestClassifier(sklearn.svm.SVC(**params)))

    self._estimator.fit(X, y)
    # `fit` has to return `self` for method chaining to work
    return self
```

So `CPUModel` uses `self._estimator` to do all the hard work and forwards methods and attributes to it. All the logic about what kind of (combined) estimator we need is then just in `fit` instead of spread out across the code that needs to know which attribute to use (`self.prob_svc` vs `self.multi_class_model` vs neither of them).

Reply:
The `cpuModelSVC` aims at reproducing the inner-working of the cuML SVC estimator. The naming is also kept identical as to reduce possible confusion for the implementation of the function in charge of transferring attributes and to ease later development. The `cpuModelSVC` inherits from the SVC model to implement the binary case (as in cuML), but also to allow the handling of hyperparameters when the CPU class is instantiated in the UniversalBase class code.